queue/redis/client_test.go

```diff
@@ -0,0 +1,124 @@
+//go:build integration
+// +build integration
+
+package redis
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/frain-dev/convoy"
+	"github.com/frain-dev/convoy/config"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/frain-dev/convoy/queue"
+	"github.com/go-redis/redis/v8"
+	"github.com/google/uuid"
+	"github.com/vmihailenco/taskq/v3"
+)
+
+func TestRedisQueue(t *testing.T) {
+	for scenario, fn := range map[string]func(t *testing.T){
+		"redis queue write":    testWritetoQueue,
+		"redis start consumer": testStartConsumer,
+	} {
+		t.Run(scenario, func(t *testing.T) {
+			fn(t)
+		})
+	}
+}
```

Comment:
i don't know if this kind of tabling will allow us efficiently test various scenarios.

Please correct me if I'm wrong.

Reply:
Can you clarify your concerns?

Reply:
oh i mean, i mean table test scenarios for testWritetoQueue for example, like [here](https://github.com/frain-dev/convoy/blob/main/datastore/bolt/application_test.go)

Reply:
Yeah, i was doing something like this initially `https://github.com/frain-dev/convoy/pull/342/commits/6047d604aad3f20837764809d438aa7979423dd5` but @Subomi requested a different approach where the tests are separated into unique functions and then called in the test table.

Reply:
@ogbanugot This is not how we write table tests in the codebase. You can look through tests in the `server` package to understand how to correct it.

Reply:
Okay, will make changes.